src/vs/workbench/contrib/bulkEdit/browser/bulkFileEdits.ts

```diff
@@ -163,18 +163,22 @@ class DeleteOperation implements IFileOperation {
 			return new Noop();
 		}
 
-		let contents: VSBuffer | undefined;
+		let fileContent: IFileContent | undefined;
 		if (!this._undoesCreateOperation && !this.options.folder) {
 			try {
-				contents = (await this._fileService.readFile(this.oldUri)).value;
+				fileContent = (await this._fileService.readFile(this.oldUri));
 			} catch (err) {
 				this._logService.critical(err);
 			}
 		}
 
 		const useTrash = this._fileService.hasCapability(this.oldUri, FileSystemProviderCapabilities.Trash) && this._configurationService.getValue<boolean>('files.enableTrash');
 		await this._workingCopyFileService.delete([this.oldUri], { useTrash, recursive: this.options.recursive, ...this.undoRedoInfo });
-		return this._instaService.createInstance(CreateOperation, this.oldUri, this.options, { isUndoing: true }, contents);
+
+		if (typeof this.options.maxSize === 'number' && fileContent && (fileContent?.size > this.options.maxSize)) {
```

Comment:
Suggestion: move to two new private functions so it is reusable and easier to read e.g.

`if (this->hasMaxFileSize() && this._doesSizeExceedMax(fileContent)) {
    return new Noop();
}`